subsys/mgmt/mcumgr/Kconfig

```diff
@@ -26,6 +26,84 @@ config APP_LINK_WITH_MCUMGR
 
 rsource "lib/Kconfig"
 
+config MGMT_MAX_MAIN_MAP_ENTRIES
+	int "Number of predicted maximum entries to main response map"
+	default 15
+	help
+	  This is number of predicted entries in main response map,
+	  the one that encapsulates everything within response.
+	  This value is used by zcbor to predict needed map encoding,
+	  and does not affect memory allocation or usage.
+	  Builtin command processors rarely add large amounts of
+	  data directly to main map, creating sub-maps instead so
+	  the default value works fine with them.
+	  If your app directly adds fields to main map, without
+	  encapsulating them, you may want to increase this value
+	  in case when encoding starts to fail.
+
+config MGMT_MIN_DECODING_LEVELS
+	int
+	help
+	  Minimal decoding levels, map/list encapsulation, required
+	  to be supported by zcbor decoding of SMP responses
+	  is auto genereated from MGMT_MIN_DECODING_LEVEL_? options.
+	  A group or command that adds additional maps/lists above the
+	  base map, which is already taken into account, should
+	  select one of the MGMT_MIN_DECODING_LEVEL_?.
+	default 0 if !MGMT_MIN_DECODING_LEVEL_1	&&	\
+		     !MGMT_MIN_DECODING_LEVEL_2 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_3 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_4 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_5
+	default 1 if MGMT_MIN_DECODING_LEVEL_1	&&	\
+		     !MGMT_MIN_DECODING_LEVEL_2 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_3 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_4 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_5
+	default 2 if MGMT_MIN_DECODING_LEVEL_2 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_3 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_4 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_5
+	default 3 if MGMT_MIN_DECODING_LEVEL_3 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_4 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_5
+	default 4 if MGMT_MIN_DECODING_LEVEL_4 &&	\
+		     !MGMT_MIN_DECODING_LEVEL_5
+	default 5 if MGMT_MIN_DECODING_LEVEL_5
```

Comment:
Would this be tidier/easier to maintain with new levels if it was created the other way around?
```
 	default 5 if MGMT_MIN_DECODING_LEVEL_5
 	default 4 if MGMT_MIN_DECODING_LEVEL_4
 	default 3 if MGMT_MIN_DECODING_LEVEL_3
 	default 2 if MGMT_MIN_DECODING_LEVEL_2
 	default 1 if MGMT_MIN_DECODING_LEVEL_1
 	default 0
```

Reply:
Fixed